server/models/user.js

```diff
@@ -37,22 +37,38 @@ UserSchema.plugin(uniqueValidator, {
 UserSchema.pre('save', function(next) {
   const password = this.password;
   if (password) {
-    bcrypt.hash(password, 10)
-      .then((hash) => {
+    bcrypt
+      .hash(password, 10)
+      .then(hash => {
         this.password = hash;
         next();
       })
-      .catch((err) => next(err))
+      .catch(err => next(err));
   } else {
     next();
   }
 });
 
+UserSchema.statics.authenticate = function(email, password) {
+  return User.findOne({ email })
+    .exec()
```

Comment:
Is `.exec()` needed here or does `.findOne()` return a promise? I'm genuinely curious.

Reply:
Still not totally clear on `.then()` vs `.exec()`, but it does seem like `.exec()` is needed with mongoose to return an actual promise.  The mongoose docs are a bit confusing, let me know what you think:

>A query also has a .then() function, and thus can be used as a promise.

But then,

>Mongoose queries are not promises. They have a .then() function for co and async/await as a convenience. However, unlike promises, calling a query's .then() can execute the query multiple times. [...] If you need a fully-fledged promise, use the .exec() function.